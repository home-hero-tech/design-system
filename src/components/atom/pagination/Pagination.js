import React from 'react';
import PropTypes from 'prop-types';
import RPagination from 'react-js-pagination';
import css from './Pagination.module.scss';

const Pagination = ({ activePage, itemsCountPerPage, totalItemsCount, pageRangeDisplayed, onClick, onChange }) => {
  return (
    <div className={css['c-pagination']}>
      <RPagination
        activePage={activePage}
        itemsCountPerPage={itemsCountPerPage}
        totalItemsCount={totalItemsCount}
        pageRangeDisplayed={pageRangeDisplayed}
        onChange={onChange}
        onClick={onClick}
        itemClass="page"
      />
    </div>
  );
};

Pagination.propTypes = {
  activePage: PropTypes.number,
  itemsCountPerPage: PropTypes.number,
  totalItemsCount: PropTypes.number,
  pageRangeDisplayed: PropTypes.number,
  onClick: PropTypes.func,
  onChange: PropTypes.func
};

Pagination.defaultProps = {
  activePage: 1,
  itemsCountPerPage: 20,
  totalItemsCount: null,
  pageRangeDisplayed: 5,
  onClick: f => f,
  onChange: f => f
};

export default Pagination;
