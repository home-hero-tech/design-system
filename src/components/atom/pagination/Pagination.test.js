import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Pagination from './Pagination';

describe('Pagination', () => {
  it('should match with the snapshot', () => {
    const wrapper = shallow(
      <Pagination activePage={1} totalItemsCount={450} pageRangeDisplayed={5} onChange={() => {}} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
