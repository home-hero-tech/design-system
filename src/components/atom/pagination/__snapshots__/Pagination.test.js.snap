// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should match with the snapshot 1`] = `
<div
  className="c-pagination"
>
  <Pagination
    activePage={1}
    firstPageText="«"
    getPageUrl={[Function]}
    hideFirstLastPages={false}
    innerClass="pagination"
    itemClass="page"
    itemsCountPerPage={20}
    lastPageText="»"
    nextPageText="⟩"
    onChange={[Function]}
    onClick={[Function]}
    pageRangeDisplayed={5}
    prevPageText="⟨"
    totalItemsCount={450}
  />
</div>
`;
